File: Domain/EduComponent.cs
namespace Domain;
public class EduComponent
{
	public int Id { get; set; }
	public string Name { get; set; }
	public string Info { get; set; }
	public int EctsCredits { get; set; }
	public bool IsOptional { get; set; }
	public Specialty Specialty { get; set; }
}

File: client-app/src/app/models/eduComponent.ts
export interface EduComponent {
	id: string;
	name: string;
	info: string;
	isOptional: boolean;
	ectsCredits: number;
}

export class EduComponentFormValues {
	id: string = "";
	name: string = "";
	info: string = "";
	isOptional: boolean = false;
	ectsCredits: number = 0;
}